### ISSUE TITLE
Missing annotations for ChannelIndex in BlackrockIO

### ISSUE THREAD

**Issue by bjoern1001001** [NeuralEnsemble/python-neo#425]
In earlier versions of BlackRockIO every ChannelIndex contained a lot of annotations, like 'nsx_hi_freq_corner' or 'nev_dig_factor'. Now they are not available any more. When trying to implement them again I noticed there was no space for them in self.raw_annotations in BaseRawIO and thus they were not passed to the object on its creation. In order to restore the annotations as they were, BaseRawIO would have to be changed to create room for the ChannelIndex annotations in self.raw_annotations and also BaseFromRawIO would have to be changed to group the annotations according to the corresponding AnalogSignals and to pass the annotations on to the created objects.
What do you think about that, @samuelgarcia? Do you think BaseFromRaw and BaseRaw should be updated to allow for this (not only for BlackRockIO but also all other IOs) or should this be implemented specifically for the BlackRockIO?
If you want to have a look at what I meant see https://github.com/bjoern1001001/python-neo/tree/fix/chidx_annotations

**Comment by samuelgarcia** [comment]
You can put annotations for ChannelIndex here
```raw_annotations['signal_channels'][channel_index]```
at BaseFomRaw they should be put in channel index when prefered_signal_group_mode = 'split-all'

In actual BlackrowRawIO you should add at line 356 something like:
```python
for c in range(sig_channels.size):
    ann = self.raw_annotations['signal_channels'][c]
    ann['nsx_hi_freq_corner'] = XXX
    ann['nev_dig_factor'] = XXX
```

Raw annotations is a big nested dict that should cover all objects
```python
raw_annotations['blocks'][block_index]# annotate Block
raw_annotations['blocks'][block_index]['segments'][seg_index]# annotate Segment
raw_annotations['blocks'][block_index]['segments'][seg_index]['signals'][channel_index]#annotate AnalogSignal
raw_annotations['blocks'][block_index]['segments'][seg_index]['units'][unit_index]#annotate SpikeTrain
raw_annotations['blocks'][block_index]['segments'][seg_index]['events'][ev_chan]#annotate Event or Epoch
raw_annotations['signal_channels'][channel_index]# annotate ChannelIndex (when 'split-all')
raw_annotations['unit_channels'][channel_index]# annotate Unit
raw_annotations['events_channels'][channel_index]# annotate nothing because do no exist in neo.core
```

To have a look at your code it would be very easier to me if you would make PR to be able to directly see the patch.

Thank you for working so much to make the link between old and new BlackrockIO.

**Comment by bjoern1001001** [comment]
Thanks a lot for the answer, so it's simpler than I thought. I did it in a similar way but did not realise that signal_channels was supposed to be for ChannelIndex. This fits quite well to your suggested changes in my pull request #426 because I had to change BaseFromRawIO to pass annotations for all objects. I adapted my code according to your suggestions and will create a pull request for this soon.
